lib/melinis/task.rb

```diff
@@ -68,7 +72,9 @@ def wrapup
     def run
       success, total = 0, 0
       begin
-        @current_run = Melinis::TaskProcessing.create!({:task_id => @task.id})
+        @current_run = Melinis::TaskProcessing.find_by(id: @melinis_task_processing_id)
```

Comment:
Why would you do it this way? You need to check if run_id is present & then do a find. Else, do a create. No need to run 2 db queries.